sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsHostMemoryStore.scala

```diff
@@ -71,12 +71,10 @@ class RapidsHostMemoryStore(
     (buffer, true)
   }
 
-  override protected def createBuffer(
-      other: RapidsBuffer,
+  override protected def createBuffer(other: RapidsBuffer, otherBuffer: MemoryBuffer,
       stream: Cuda.Stream): RapidsBufferBase = {
     val (hostBuffer, isPinned) = allocateHostBuffer(other.size)
```

Comment:
We leak `otherBuffer` if this throws.  Should start the method with `withResource(otherBuffer) { _ =>` to ensure it is always closed.